edx_rest_api_client/client.py

```diff
@@ -10,70 +11,117 @@
 from edx_rest_api_client.__version__ import __version__
 
 
-class EdxRestApiClient(slumber.API):
-
-    @classmethod
-    def user_agent(cls):
-        """
-        Return a User-Agent that identifies this client.
-
-        Example:
-            python-requests/2.9.1 edx-rest-api-client/1.7.2 ecommerce
-
-        The last item in the list will be the application name, taken from the
-        OS environment variable EDX_REST_API_CLIENT_NAME. If that environment
-        variable is not set, it will default to the hostname.
+def user_agent():
+    """
+    Return a User-Agent that identifies this client.
+
+    Example:
+        python-requests/2.9.1 edx-rest-api-client/1.7.2 ecommerce
+
+    The last item in the list will be the application name, taken from the
+    OS environment variable EDX_REST_API_CLIENT_NAME. If that environment
+    variable is not set, it will default to the hostname.
+    """
+    client_name = 'unknown_client_name'
+    try:
+        client_name = os.environ.get("EDX_REST_API_CLIENT_NAME") or socket.gethostbyname(socket.gethostname())
+    except:  # pylint: disable=bare-except
+        pass  # using 'unknown_client_name' is good enough.  no need to log.
+    return "{} edx-rest-api-client/{} {}".format(
+        requests.utils.default_user_agent(),  # e.g. "python-requests/2.9.1"
+        __version__,  # version of this client
+        client_name
+    )
+
+USER_AGENT = user_agent()
+
+
+def get_oauth_access_token(url, client_id, client_secret, token_type='jwt', grant_type='client_credentials'):
+    """ Retrieves OAuth 2.0 access token using the client credentials grant.
+
+    Args:
+        url (str): Oauth2 access token endpoint
+        client_id (str): client ID
+        client_secret (str): client secret
+        token_type (str): Type of token to return. Options include bearer and jwt.
+
+    Returns:
+        tuple: Tuple containing access token string and expiration datetime.
+    """
+    now = datetime.datetime.utcnow()
+
+    response = requests.post(
+        url,
+        data={
+            'grant_type': grant_type,
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'token_type': token_type,
+        },
+        headers={
+            'User-Agent': USER_AGENT,
+        },
+    )
+
+    data = response.json()
+
+    try:
+        access_token = data['access_token']
+        expires_in = data['expires_in']
+    except KeyError:
+        raise requests.RequestException(response=response)
+
+    expires_at = now + datetime.timedelta(seconds=expires_in)
+
+    return access_token, expires_at
+
+
+class EdxSession(requests.Session):
```

Comment:
@robrap Want to help us with naming?  What are your thoughts on using something other than `Session` here?  This is essentially a helper SDK class for folks to use to call our edX APIs using OAuth+JWT.

Reply:
How about `OAuthJWTClient` or `OAuthAPIClient`?  
We don't necessarily need `Edx` in the name since it's part of an edX repo and there's nothing here that is really edX protocol specific, right?
Also, is there anything here specifically about JWTs?   If not, it can just be `OAuthAPIClient`.

Reply:
- I'm not sure that a subclass of `requests.Session` that avoids the word `Session` makes sense in terms of avoiding confusion.  @davestgermain provided the `OAuth2Session` example in this other comment: https://github.com/edx/edx-rest-api-client/pull/35#issuecomment-431120636.  - - There are several comments in this code that use the term `JWT`.  Are we planning on removing those references from comments, or maybe using `for example` language, or is there something JWT specific?

What about `OAuthClientSession`?  Some permutation of these words needs to work. ;)

Reply:
Maybe just find a way to clarify the intention stated above in the comments in case there is ever a need to enhance in the future?

Reply:
I believe the fact that JWTs are transferred during the API handshake is opaque to the API client.  From the API client's perspective, they simply make the call to an API.  We do not expose nor plan to expose contents of the JWT to the API client, right?

I think having a name that better suits its purpose is worthwhile - especially if it is housed in a common shared library for others to use.  If we can't come up with a better name, then you can go with `OAuthAPIClient`.